controllers/capa_utils.go

```diff
@@ -41,3 +51,46 @@ func isRoleUsedElsewhere(ctx context.Context, ctrlClient client.Client, roleName
 
 	return false, err
 }
+
+func removeFinalizer(ctx context.Context, k8sClient client.Client, object client.Object, role string) error {
+	logger := log.FromContext(ctx)
+
+	if !controllerutil.ContainsFinalizer(object, key.FinalizerName(role)) {
+		logger.Info("finalizer already removed")
+		return nil
+	}
+
+	for i := 1; i <= maxPatchRetries; i++ {
```

Comment:
[nit]

`maxPatchRetries` should be called `maxPatchAttempts`.

This is about wording: "total number of attempts" = "first attempt + N retries". So the number of retries is always 1 less than the total number of attempts.

Reply:
done